MetaMask/metamask-mobile
# chore: Improve Keyring/Accounts error handling and logs

## **Description**
This PR does the following

1. unmasks non sensitive data from the Accounts Controller and Keyring controller to help us better debug issues in prod. This means that more data will be provided in the sentry state logs. I have ensured that all addresses have remained masked
2. adds logs when creating key components such as the engine and the accounts controller. This will help us debug issues with empty state.
3. adds validation to the migrations for key components. Currently we are validating that the accounts controller and keyring controller data are sound. This validation will not block the app from initializing but will log the errors it finds.

## **Related issues**

Progresses : https://github.com/MetaMask/metamask-mobile/issues/12408

## **Manual testing steps**

1. this change only adds logs and error handling to the app so there should be no functional changes. that being said, we should still verify that the upgrade path is working.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
